dashboard-app/src/hooks/useAxiosErrors.ts

```diff
@@ -0,0 +1,21 @@
+import { Dispatch, SetStateAction, useEffect } from 'react';
+import { AxiosError } from 'axios';
+import { AggregatedData } from '../dashboard/dataManipulation/logsToAggregatedData';
+import { Dictionary } from 'ramda';
+import { useResettableState } from './useResettableState';
+
+export const useAxiosErrors = (
+  error: AxiosError | undefined,
+  data: AggregatedData | undefined) => {
+
+  const [errors, setErrors] = useResettableState<Dictionary<string>>({}, [data]);
+
+  // set errors on response error
+  useEffect(() => {
+    if (error) {
+      setErrors({ data: error.message });
+    }
+  }, [error]);
+
+  return [errors, setErrors] as [Dictionary<string>, Dispatch<SetStateAction<Dictionary<string>>>];
```

Comment:
Not sure this interface makes sense. You're allowing the caller to set the `errors` state directly via returning `setErrors`, but also updating it automatically everytime `error: AxiosError` changes.

I think it should probably look either like my example in my previous comment, or just don't return `setErrors`. If this hook is only for a specific pattern, then maybe the latter is fine.

Reply:
I don't understand your implementation above. What would it look like in the page element?

Reply:
Well you'd have to lift the `useEffect` call out into the parent, so probably your version is better so we can avoid that. I guess just don't return `setErrors`, unless you were thinking of a particular use case that I can't see?

Reply:
That won't work. `setErrors` needs to be passed to any function on the page that is doing error handling. 

I think my imagining of `useAxiosErrors` is different to yours. I was imagining it a fancy custom useState hook that has bespoke special useEffect magic. A bit like a higher order component but for hooks, the `setState` still has it's standard state setting functionality but ✨ *_upgraded_* ✨

Here's what it can do:
- able to set initial state
- able to reset state on change of 
- able to be updated as an when we want

In app use case:
- errors set to empty object on initial load
- if any request errors it updates the error from the request
- if any manipulation errors (table/graph/csv parsing) happen it is updated
- if any new request is made it wipes all errors
	- this is because all manipulation errors are based off of the request data. If new data is requested, it is assumed that the previous error is no longer relevant.

3rd bullet point is why we need to pass out `setErrors` so that data manipulation functions can update errors accordingly

Reply:
Ah ok, so we want this to be used not only for Axios errors then. Maybe the name I suggested is misleading, do you have a better idea for one?

Reply:
-`useErrors`
-`setAndUseErrors`
-`setUseResetErrors`

Reply:
Lets go with `useErrors` for now then.